In [5]:
import numpy as np

In [6]:
Lbox = 40  #size of the box
ref = np.array([3,5,23])   # reference coordinate (x0, y0 , z0)

In [7]:
Ntot = 1000

FAKE_DATA = Lbox * np.random.rand(Ntot, 3) 

# FIND THE DISTANCES OF ALL THE POINTS IN FAKE_DATA to ref

In [8]:
distance = np.sum((ref[None,:]-FAKE_DATA)**2  ,  axis = 1 )**0.5

# FIND THE NUMBER OF POINTS THAT ARE WITHIN [3,5] TO THE REFERENCE POINT

In [80]:
#print distance[3]

20.81727063554765


In [10]:
np.max(distance)

51.95670710056746

In [11]:
np.min(distance)

2.6596530532635403

In [12]:
A =(5 < distance) & ( distance < 15)

In [13]:
#print A

In [14]:
distance[A]

array([ 9.13585865,  9.09299305, 14.25030779, 12.13498147, 13.51242372,
       11.56154811,  9.07663245, 14.22863078, 13.76265579, 14.50283527,
        6.919153  , 12.29427758,  9.62445766, 14.06402375, 13.05230804,
       13.66440734, 10.21500485, 14.19653795, 14.34268415, 10.38865996,
        7.54987052, 13.38462399, 11.64969726,  6.31177939,  7.66353158,
        9.37308138, 10.54977497,  6.97664798,  9.70331641, 14.60380511,
       12.15929573, 13.000354  , 13.7950787 , 12.82856333,  7.6425908 ,
        8.74735257,  8.31790856, 12.87537898, 13.04577709, 12.33883817,
       13.99690453, 10.98297866,  8.50592106,  9.21075088, 11.94970643,
       14.25114659,  9.7845237 , 12.58698726, 10.65312733, 13.8839309 ,
        5.02300076, 10.03124059,  5.8042558 , 14.65084451, 10.50854133,
       14.20427661, 14.71748119, 13.97023645,  6.01486187,  9.34662384,
       12.26640665, 11.70029425, 14.36540261, 13.65265106, 12.59375168,
       11.77500671, 13.53239248, 11.46723077, 14.57612036, 12.68

In [15]:
len(distance[A])

84

# FIND THE AVERAGE NUMBER OF POINTS THAT ARE WITHIN [3,5] FROM OTHER POINTS

In [36]:
zebra = []

for point in FAKE_DATA:
    
    distance_to_point = np.sum((point[None,:]-FAKE_DATA)**2  ,  axis = 1 )**0.5 #distance from all the points in fake data to the point
    A = (5 < distance_to_point) & ( distance_to_point < 15)
    x = len(distance_to_point[A])
    zebra.append(x)

In [40]:
zebra  = np.array(zebra)

In [41]:
zebra

array([ 59, 111, 145, 152, 129, 115, 177, 110, 157, 149, 101, 160,  70,
        99,  90, 103,  77, 118,  91, 103, 134, 119, 114,  91,  92, 159,
       110, 194, 179, 119,  91, 129,  95, 109, 115,  93, 140, 187,  72,
       152, 171,  88,  60,  83, 111, 121, 143,  84, 151, 126, 136, 134,
       179, 138, 142, 213, 161, 203, 126,  79, 164, 116, 193,  73,  73,
        99, 172, 116, 178,  51,  83,  98,  69, 124, 151, 125,  67, 102,
       119, 199, 144,  99, 168, 106,  97, 101, 204,  48, 178, 106, 191,
       181,  68,  70,  70, 135, 150,  88,  74, 192, 120, 108, 181, 181,
       198,  63,  37, 117, 161, 112, 105, 108, 104, 195,  80,  96, 125,
       121,  83, 115,  89, 163,  92, 119,  84, 120, 163, 166, 124,  94,
        73, 126, 120,  91, 100, 110, 199, 125,  76, 181, 147, 184, 188,
        97, 155, 104, 124,  85, 101, 113,  80, 142, 166, 109, 184, 131,
       145, 113,  59, 101, 193, 134, 140, 124, 194,  77, 189, 155, 100,
       166, 122, 172, 112, 203,  82, 155, 110,  36, 190, 126, 16

WHAT IS IN ZEBRA? [59, 111, ...]

That means there are 59 points between 5 and 15 Mpcs away from the first point, and there 111 points between 5 and 15 
Mpcs away from the second point, ... 

In [43]:
np.mean(zebra) # this is the average number of points that are between 5 and 15 Mpcs away from other points in our data

125.81

# introduction to lists

In [31]:
l = []
l.append(1)
print l

[1]


In [32]:
l.append(2)

In [34]:
l.append(-10)

In [35]:
l

[1, 2, -10]

In [38]:
np.array(l)

array([  1,   2, -10])